### PR TITLE
Feature/workflow approval thresholds

### PR DESCRIPTION
**Pull request by nathanwoulfe**
Adds/updates docs for new Workflow feature: Approval thresholds. No release date set yet, will update here when that is confirmed. Given, this is a draft for now.

I haven't updated the images in the workflow settings docs - wondering if these are necessary as there's now too many settings to fit into a single image (and more to come with a couple of additional configuration options). Would the text-based list be enough?

**RC Release:** April 26th
**Release:** May 9th

### CODE REVIEW

**Comment by sofietoft** [comment]
Thanks for the PR @nathanwoulfe !
I'll review it as soon as you mark it as "Ready for Review".

Remember that **new articles need to be added to the SUMMARY.md file in order to show up on the live docs.**
Also, this is only added to the Workflow v10 docs - is that intentional?

**Comment by nathanwoulfe** [comment]
Heya @sofietoft, I'll add the v11 docs once I've finalised the v10, easier than trying to keep both in sync. Plan is RC next week, so will leave you plenty of time for review.

**Comment by sofietoft** [comment]
Ah, that makes sense 😄 Thanks!
It would be great if you could mark it as "Ready for Review" in good time before the RC release, so I have time to review it.
Preferably before the end of the week 🙏

File: 10/umbraco-workflow/getting-started/approval-thresholds.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Approval thresholds
+
+{% hint style="info" %}
+This feature requires a license - learn more about [Workflow's licensing model](https://umbraco.com/products/umbraco-workflow)
```
Comment (sofietoft) [inline 10/umbraco-workflow/getting-started/approval-thresholds.md:4]: We have an article about the licensing in the docs as well: https://docs.umbraco.com/umbraco-workflow/licensing 
Does it make sense to link to that instead perhaps?

File: 10/umbraco-workflow/getting-started/approval-thresholds.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+Workflow's default behaviour requires one member of an approval group to approve a pending task to advance it through the workflow. 
+
+The approval thresholds feature introduces configuration options to set the required number of approvals for a given workflow stage.
+
+The threshold options are:
+
+- **One**: Pending tasks require approval from any member of the assigned approval group. This is the default behaviour for all installations (trial and licensed).
+- **Most**: Pending tasks require approcal from an absolute majority of group members. For example, a group with three members requires two approvals and a group with four members requires three approvals.
+- **All**: Pending tasks require approval from all group members.
```
Comment (sofietoft) [inline 10/umbraco-workflow/getting-started/approval-thresholds.md:15]: How about we put these into a table instead?
I think that would make it a little easier to read.

File: 10/umbraco-workflow/getting-started/approval-thresholds.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+The workflow detail UI displays the following:
+
+- The status of the current task.
+- Approval status for members of the current group.
+- Progress towards meeting the threshold for the current approval stage.
+- Future tasks and the assigned group and its users.
```
Comment (sofietoft) [inline 10/umbraco-workflow/getting-started/approval-thresholds.md:22]: Sounds like this is two things? But I might be wrong.

```suggestion
- Future tasks
- The assigned group and its users.
```

File: 10/umbraco-workflow/getting-started/approval-thresholds.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+Approval thresholds are managed via settings in the Workflow section or in the `appsettings.json` file. Node-specific thresholds are set directly on the workflow configuration in the Workflow content app.
+
+### Settings
+
+The settings below can be set from the Backoffice or via settings customization in the `appsettings.json` file (refer to [Settings customization](https://docs.umbraco.com/umbraco-workflow/getting-started/configuration#settingscustomization) for implementation instructions).
+
+- **Approval threshold**: Sets the global approval threshold to One, Most or All.
+- **Rejection resets approvals**: When `true`, and the approval threshold is Most or All, rejecting a task resets the previous approvals for the workflow stage.
+- **Allow configuring approval threshold**: When `true`, enables setting the approval threshold for any stage of a workflow (on a content node or Document Type).
```
Comment (sofietoft) [inline 10/umbraco-workflow/getting-started/approval-thresholds.md:43]: How about we put these into a table instead?
I think that would make it a little easier to read.

File: 10/umbraco-workflow/getting-started/configuration.md [hunk 5]
```diff
@@ -140,6 +143,22 @@ Sets the workflow type to one of Explicit (0), Implicit (1), or Exclude (2):
 - Implicit auto-approves steps where the author is a member of the approving group
 - Exclude behaves similar to Explicit, but excludes the original author from any notifications (that is the author can not approve their own work)
 
+#### ApprovalThreshold (int, requires a license)
+
+Sets the default approval threshold to one of One (0), Most (1), or All (2):
+
+- One: pending task requires approval from any member of the assigned approval group. This is the default behaviour for all installations (trial and licensed)
+- Most: pending task requires an absolute majority of group members. For example, a group with three members requires two approvals, a group with four members requires three approvals, and a group with six members requires four approvals.
+- All: pending task requires approval from all group members.
```
Comment (sofietoft) [inline 10/umbraco-workflow/getting-started/configuration.md:152]: Again, how about we put these into a table instead?
I think that would make it a little easier to read.
We could also add a column for the int value.

File: 10/umbraco-workflow/release-notes.md [hunk 6]
```diff
@@ -12,10 +12,14 @@ In this section, you can find the release notes for each version of Umbraco Work
 
 <summary>Version 10</summary>
 
+### [10.2.0](https://github.com/umbraco/Umbraco.Workflow.Issues/issues?q=is%3Aissue+is%3Aclosed+label%3Arelease%2F10.2.0) (May 9th 2023)
+
+* **FEATURE** => Introduces approval thresholds
```
Comment (sofietoft) [inline 10/umbraco-workflow/release-notes.md:17]: Should we add a link to the new article here, perhaps?

**Review by sofietoft — COMMENTED** [review]
I've completed the review.
Made a few comments here and there - let me know what you think!

File: 10/umbraco-workflow/getting-started/approval-thresholds.md (hunk 1 above)
Comment (sofietoft) [reply 10/umbraco-workflow/getting-started/approval-thresholds.md:4]: Let's keep the link to .com.

File: 10/umbraco-workflow/getting-started/approval-thresholds.md (hunk 2 above)
Comment (sofietoft) [reply 10/umbraco-workflow/getting-started/approval-thresholds.md:15]: I'll add the tabs once the PR has been merged.

**Review by sofietoft — APPROVED** [review]

**Pull request merged**